BUG: `bot.send_to` 发送 MediaGroup 失败

## 环境
Windows 11 22H2
Python 3.10.11
nonebot2==2.0.0
nonebot-adapter-telegram==0.1.0b13

## 问题描述
发送多张图片时失败

```python3

from nonebot import on_command

from nonebot.adapters.telegram import Bot as TelegramBot, Event as TelegramEvent
from nonebot.adapters.telegram.message import File


@on_command("group_media_test").handle()
async def _(bot: TelegramBot, event: TelegramEvent):
    await bot.send(
        event,

        File.photo(r'https://files.yande.re/jpeg/9ce83f944ff73e7cb5667eb1cc61df09/yande.re%201082595%20azur_lane%20feet%20loli%20pantsu%20pantyhose%20see_through%20skirt_lift%20thong%20unicorn_%28azur_lane%29%20wuti_%28qq1432872407%29.jpg') +
        File.photo(r'C:\Users\Ailit\Pictures\01164-1638397549.png') +
        'This is a GroupMedia' +
        File.photo(r'C:\Users\Ailit\Pictures\201809 (2).jpg') +
        File.photo(r'AgACAgUAAxkBAAPqZJ3KtVqM9P4_C4rid3Yd1-YAAaLqAAJDtzEbjq_pVKUxcDlGJM2JAQADAgADeQADLwQ')
    )
```

日志如下

```python3
06-30 02:22:18 [DEBUG] nonebot | Telegram | Calling API sendMediaGroup
06-30 02:22:20 [INFO] nonebot | Matcher(type='message', module=test.v2.telegram_test) running complete
06-30 02:22:20 [ERROR] nonebot | Running Matcher(type='message', module=test.v2.telegram_test) failed.
Traceback (most recent call last):
  File "C:\Users\Ailit\PycharmProjects\nonebot2_miya\bot.py", line 59, in <module>
    nonebot.run()
  File "C:\Users\Ailit\PycharmProjects\nonebot2_miya\venv\lib\site-packages\nonebot\__init__.py", line 309, in run
    get_driver().run(*args, **kwargs)
  File "C:\Users\Ailit\PycharmProjects\nonebot2_miya\venv\lib\site-packages\nonebot\drivers\fastapi.py", line 198, in run
    uvicorn.run(
  File "C:\Users\Ailit\PycharmProjects\nonebot2_miya\venv\lib\site-packages\uvicorn\main.py", line 569, in run
    server.run()
  File "C:\Users\Ailit\PycharmProjects\nonebot2_miya\venv\lib\site-packages\uvicorn\server.py", line 60, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "C:\Users\Ailit\AppData\Local\Programs\Python\Python310\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "C:\Users\Ailit\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 636, in run_until_complete
    self.run_forever()
  File "C:\Users\Ailit\AppData\Local\Programs\Python\Python310\lib\asyncio\windows_events.py", line 321, in run_forever
    super().run_forever()
  File "C:\Users\Ailit\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 603, in run_forever
    self._run_once()
  File "C:\Users\Ailit\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 1909, in _run_once
    handle._run()
  File "C:\Users\Ailit\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\Ailit\PycharmProjects\nonebot2_miya\venv\lib\site-packages\nonebot\message.py", line 467, in check_and_run_matcher
    await _run_matcher(
> File "C:\Users\Ailit\PycharmProjects\nonebot2_miya\venv\lib\site-packages\nonebot\message.py", line 419, in _run_matcher
    await matcher.run(bot, event, state, stack, dependency_cache)
  File "C:\Users\Ailit\PycharmProjects\nonebot2_miya\venv\lib\site-packages\nonebot\internal\matcher\matcher.py", line 753, in run
    await self.simple_run(bot, event, state, stack, dependency_cache)
  File "C:\Users\Ailit\PycharmProjects\nonebot2_miya\venv\lib\site-packages\nonebot\internal\matcher\matcher.py", line 728, in simple_run
    await handler(
  File "C:\Users\Ailit\PycharmProjects\nonebot2_miya\venv\lib\site-packages\nonebot\dependencies\__init__.py", line 108, in __call__
    return await cast(Callable[..., Awaitable[R]], self.call)(**values)
  File "C:\Users\Ailit\PycharmProjects\nonebot2_miya\test\v2\telegram_test.py", line 19, in _
    await bot.send(
  File "C:\Users\Ailit\PycharmProjects\nonebot2_miya\venv\lib\site-packages\nonebot\adapters\telegram\bot.py", line 279, in send
    return await self.send_to(
  File "C:\Users\Ailit\PycharmProjects\nonebot2_miya\venv\lib\site-packages\nonebot\adapters\telegram\bot.py", line 243, in send_to
    return await self.send_media_group(
  File "C:\Users\Ailit\PycharmProjects\nonebot2_miya\venv\lib\site-packages\nonebot\adapters\telegram\bot.py", line 103, in call_api
    sign.return_annotation, await super().call_api(api, **kargs)
  File "C:\Users\Ailit\PycharmProjects\nonebot2_miya\venv\lib\site-packages\nonebot\internal\adapter\bot.py", line 120, in call_api
    raise exception
  File "C:\Users\Ailit\PycharmProjects\nonebot2_miya\venv\lib\site-packages\nonebot\internal\adapter\bot.py", line 98, in call_api
    result = await self.adapter._call_api(self, api, **data)
  File "C:\Users\Ailit\PycharmProjects\nonebot2_miya\venv\lib\site-packages\nonebot\adapters\telegram\adapter.py", line 235, in _call_api
    raise ActionFailed(json.loads(response.content)["description"])
nonebot.adapters.telegram.exception.ActionFailed: <ActionFailed Bad Request: failed to send message #2 with the error message "IMAGE_PROCESS_FAILED">
```

看了一下 `bot.send_to` 中关于生成 MediaGroup 的部分, 发现传入的即使是文件路径, 经 `InputMedia` parsed 后在会变成 `bytes`, 并由此将该 `bytes` 直接当作图片上传而直接导致了上述问题

![image](https://github.com/nonebot/adapter-telegram/assets/41713304/2bf932bd-e6f4-49b9-b67e-654c510efc26)

考虑到 Telegram API 文档关于 [inputmedia](https://core.telegram.org/bots/api#inputmedia) 中 `media` 参数的描述：

> File to send. Pass a file_id to send a file that exists on the Telegram servers (recommended), pass an HTTP URL for Telegram to get a file from the Internet, or pass “attach://<file_attach_name>” to upload a new one using multipart/form-data under <file_attach_name> name.

发送文件的 file_id, url 和文件路径应当是比文件内容更优先解析的类型

https://github.com/nonebot/adapter-telegram/blob/458326060ca3955de8b310e02c67f9d2c851c412/nonebot/adapters/telegram/model.py#L784-L792

因此 789 行, `media` 的类型更改为 `Union[str, InputFile]` 似乎更为合适，且修改后上述问题得以解决

--- a/nonebot/adapters/telegram/model.py
+++ b/nonebot/adapters/telegram/model.py
@@ -786,7 +786,7 @@ InputFile = Union[bytes, Tuple[str, bytes]]
 
 class InputMedia(BaseModel):
     type: str
-    media: Union[InputFile, str]
+    media: Union[str, InputFile]
     caption: Optional[str] = None
     parse_mode: Optional[str] = None
     caption_entities: Optional[List[MessageEntity]] = None
@@ -799,7 +799,7 @@ class InputMediaPhoto(InputMedia):
 
 class InputMediaVideo(InputMedia):
     type: Literal["video"] = "video"
-    thumbnail: Optional[Union[InputFile, str]] = None
+    thumbnail: Optional[Union[str, InputFile]] = None
     width: Optional[int] = None
     height: Optional[int] = None
     duration: Optional[int] = None
@@ -809,7 +809,7 @@ class InputMediaVideo(InputMedia):
 
 class InputMediaAnimation(InputMedia):
     type: Literal["animation"] = "animation"
-    thumbnail: Optional[Union[InputFile, str]] = None
+    thumbnail: Optional[Union[str, InputFile]] = None
     width: Optional[int] = None
     height: Optional[int] = None
     duration: Optional[int] = None
@@ -818,7 +818,7 @@ class InputMediaAnimation(InputMedia):
 
 class InputMediaAudio(InputMedia):
     type: Literal["audio"] = "audio"
-    thumbnail: Optional[Union[InputFile, str]] = None
+    thumbnail: Optional[Union[str, InputFile]] = None
     duration: Optional[int] = None
     performer: Optional[str] = None
     title: Optional[str] = None
@@ -826,7 +826,7 @@ class InputMediaAudio(InputMedia):
 
 class InputMediaDocument(InputMedia):
     type: Literal["document"] = "document"
-    thumbnail: Optional[Union[InputFile, str]] = None
+    thumbnail: Optional[Union[str, InputFile]] = None
     disable_content_type_detection: Optional[bool] = None
 
 

